test/suite/test_truncate15.py

```diff
@@ -0,0 +1,217 @@
+
+        # Reopen the connection so nothing is in memory and we can fast-truncate.
+        self.reopen_conn()
+
+        # Truncate the data at time 25, but prepare at 20 and make durable 30.
+        self.session.begin_transaction()
+        err = self.truncate(ds.uri, ds.key, nrows // 4 + 1, nrows // 4 + nrows // 2)
+        self.assertEqual(err, 0)
+        self.session.prepare_transaction('prepare_timestamp=' + self.timestamp_str(20))
+        self.session.timestamp_transaction('commit_timestamp=' + self.timestamp_str(25))
+        self.session.commit_transaction('durable_timestamp=' + self.timestamp_str(30))
+
+        # Make sure we did at least one fast-delete. For columns, there's no fast-delete
+        # support (yet) so assert we didn't.
+        stat_cursor = self.session.open_cursor('statistics:', None, None)
+        fastdelete_pages = stat_cursor[stat.conn.rec_page_delete_fast][2]
+        if self.key_format == 'r':
+            self.assertEqual(fastdelete_pages, 0)
+        else:
+            self.assertGreater(fastdelete_pages, 0)
+
+        # Advance stable.
+        self.conn.set_timestamp('stable_timestamp=' + self.timestamp_str(30))
+        self.session.checkpoint()
+
+        # Reopen the connection so nothing is in memory.
+        self.reopen_conn(".", "cache_size=1MB,eviction_dirty_target=90,eviction_dirty_trigger=100" +
+            ",eviction_updates_target=90,eviction_updates_trigger=100,readonly=true")
+
+        # Validate the data.
+        try:
+            # At time 10 we should see all value_a.
+            self.check(ds.uri, ds.key, nrows, 0, value_a, 10)
+            #self.evict_cursor(ds.uri, ds, nrows, 10)
+
+            # At time 20 we should still see all value_a.
+            self.check(ds.uri, ds.key, nrows, 0, value_a, 20)
+            #self.evict_cursor(ds.uri, ds, nrows, 20)
+
+            # At time 25 we should still see half value_a, and for FLCS, half zeros.
+            self.check(ds.uri, ds.key, nrows // 2, nrows // 2, value_a, 25)
+            #self.evict_cursor(ds.uri, ds, nrows // 2, 25)
+
+            # At time 30 we should also see half value_a, and for FLCS, half zeros.
+            self.check(ds.uri, ds.key, nrows // 2, nrows // 2, value_a, 30)
+            #self.evict_cursor(ds.uri, ds, nrows // 2, 30)
+        except WiredTigerError as e:
+            # If we get WT_ROLLBACK while reading, assume it's because we overflowed the
+            # cache, and fail. (If we don't trap this explicitly, the test harness retries
+            # the whole test, which is not what we want.)
+            if wiredtiger_strerror(WT_ROLLBACK) in str(e):
+                self.assertTrue(False)
+            else:
+                raise e
+
+        # Move the stable timestamp forward before exiting so we don't waste time rolling
+        # back the changes during shutdown.
+        self.conn.set_timestamp('stable_timestamp=' + self.timestamp_str(50))
```

Comment:
We already moved the stable timestamp to 30, the RTS removes any data if it is greater than the stable timestamp, so nothing gets removed in this case.

Reply:
Good point. I think that got pasted from another test :-)  I removed it in [9933eaa](https://github.com/wiredtiger/wiredtiger/pull/8156/commits/9933eaaedc2a3d179cec8715a13851eb52f84725).